### DISCUSSION TITLE
Understanding UCX Shuffle Plugin

### DISCUSSION THREAD

**Discussion by Nikhilpa1** [NVIDIA/spark-rapids#8173] [Q&A] [answered]
I am interested in understanding the low-level architecture of the UCX shuffle plugin. However, I'm having difficulties grasping the plugin's implementation details([_here_](https://github.com/NVIDIA/spark-rapids/tree/branch-22.10/shuffle-plugin/src/main/scala/com/nvidia/spark/rapids/shuffle/ucx)) and how it makes use of the UCX library for efficient communication. Any suggestions or references are greatly appreciated.

**Comment by sameerz** [comment, marked as answer]
Hi @Nikhilpa1 , some of our team members presented an overview of UCX here: https://www.nvidia.com/en-us/on-demand/session/gtcspring21-s31822/setFocus/ (you will need to create a free account to view).  If you would like to have a further deep dive discussion, please reach out to use at [spark-rapids-support@nvidia.com](mailto:spark-rapids-support@nvidia.com)

**Reply by Nikhilpa1** [reply]
Sure, Thanks for the info.
